components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -168,13 +170,21 @@ public void deleteBusinessRule(String uuid) throws TemplateManagerException { //
         // todo: else: If found Business Rule is from scratch
     }
 
-    public boolean deployBusinessRule() {
-        boolean successfullyDeployed = false;
-
+    public boolean deployBusinessRule(BusinessRuleFromTemplate businessRuleFromTemplate) throws TemplateManagerException {
+        Map<String, Artifact> derivedTemplates = deriveTemplates(businessRuleFromTemplate);
+        for (String templateUUID : derivedTemplates.keySet()) {
+            try {
+                deployTemplate(templateUUID, derivedTemplates.get(templateUUID));
+            } catch (TemplateManagerException e) {
+                log.error("Failed to deploy " + derivedTemplates.get(templateUUID).getType() + " : " + templateUUID, e);
+                return false;
```

Comment:
Although the method signature says throws TemplateManagerException we are catching and logging it. Shall we change the method sig?

Reply:
This was changed in 96eec5d75b6c71ffe19967e28bcc1a13c5010ac5